### PR TITLE
src/cstart: fix typo in cstart script

### PR DESCRIPTION
**Pull request by ideepika**
## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by tchaikov** [comment]
```
Reading package lists...
E: Failed to fetch https://chacra.ceph.com/r/libboost/master/1d7c7a00cc3f37e340bae0360191a757b44ec80c/ubuntu/bionic/flavors/default/dists/bionic/main/binary-amd64/Packages  404  Not Found [IP: 8.43.84.139 443]
E: Some index files failed to download. They have been ignored, or old ones used instead.
Build step 'Execute shell' marked build as failure
Setting status of 4390de26d8d1928f612ca67d45439a8379b
```

i think it's related to the data lost we have yesterday.

**Comment by tchaikov** [comment]
@ideepika is cstart.sh used anywhere ? i am wondering how we shall test this change...

**Comment by ideepika** [comment]
@tchaikov I was testing it locally, maybe I can paste output, I am not familiar if there exist any other test otherwise

**Comment by tchaikov** [comment]
> @tchaikov I was testing it locally, maybe I can paste output, I am not familiar if there exist any other test otherwise

yeah, the test result would be helpful to convince the reviewers.

**Comment by sebastian-philipp** [comment]
> > @tchaikov I was testing it locally, maybe I can paste output, I am not familiar if there exist any other test otherwise
> 
> yeah, the test result would be helpful to convince the reviewers.

cstart is somewhat similar to vstart, but a recent addition by @liewegas a few months ago.

**Comment by tchaikov** [comment]
> > > @tchaikov I was testing it locally, maybe I can paste output, I am not familiar if there exist any other test otherwise
> > 
> > 
> > yeah, the test result would be helpful to convince the reviewers.
> 
> cstart is somewhat similar to vstart, but a recent addition by @liewegas a few months ago.

@sebastian-philipp thanks for explaining the history of cstart. what i want to see is some notes from the author of the changeset explaining how the change was tested.

**Comment by tchaikov** [comment]
> > ```
> > Reading package lists...
> > E: Failed to fetch https://chacra.ceph.com/r/libboost/master/1d7c7a00cc3f37e340bae0360191a757b44ec80c/ubuntu/bionic/flavors/default/dists/bionic/main/binary-amd64/Packages  404  Not Found [IP: 8.43.84.139 443]
> > E: Some index files failed to download. They have been ignored, or old ones used instead.
> > Build step 'Execute shell' marked build as failure
> > Setting status of 4390de26d8d1928f612ca67d45439a8379b
> > ```
> > 
> > 
> > i think it's related to the data lost we have yesterday.
> 
> 
> Didn't get the context can you elaborate

i think we have an issue of data lost in our LRC that's why we have 404 when trying to pull packages from https://chacra.ceph.com/ . if you are interested in the details, please head over to `#sepia` channel.

**Comment by ideepika** [comment]
> > > ```
> > > Reading package lists...
> > > E: Failed to fetch https://chacra.ceph.com/r/libboost/master/1d7c7a00cc3f37e340bae0360191a757b44ec80c/ubuntu/bionic/flavors/default/dists/bionic/main/binary-amd64/Packages  404  Not Found [IP: 8.43.84.139 443]
> > > E: Some index files failed to download. They have been ignored, or old ones used instead.
> > > Build step 'Execute shell' marked build as failure
> > > Setting status of 4390de26d8d1928f612ca67d45439a8379b
> > > ```
> > > 
> > > 
> > > i think it's related to the data lost we have yesterday.
> > 
> > 
> > Didn't get the context can you elaborate
> 
> i think we have an issue of data lost in our LRC that's why we have 404 when trying to pull packages from https://chacra.ceph.com/ . if you are interested in the details, please head over to `#sepia` channel.



> > > ```
> > > Reading package lists...
> > > E: Failed to fetch https://chacra.ceph.com/r/libboost/master/1d7c7a00cc3f37e340bae0360191a757b44ec80c/ubuntu/bionic/flavors/default/dists/bionic/main/binary-amd64/Packages  404  Not Found [IP: 8.43.84.139 443]
> > > E: Some index files failed to download. They have been ignored, or old ones used instead.
> > > Build step 'Execute shell' marked build as failure
> > > Setting status of 4390de26d8d1928f612ca67d45439a8379b
> > > ```
> > > 
> > > 
> > > i think it's related to the data lost we have yesterday.
> > 
> > 
> > Didn't get the context can you elaborate
> 
> i think we have an issue of data lost in our LRC that's why we have 404 when trying to pull packages from https://chacra.ceph.com/ . if you are interested in the details, please head over to `#sepia` channel.

I didn't test anything yesterday but looks like a horrifying situation! I found some bugs in cstart while running, will fix and update

**Comment by ideepika** [comment]
```
[ideepika@vossi04 build]$ ../src/cstart.sh 
fsid 37973791-2508-40fe-85f5-f46afad11078
shortid 37973791
ip 172.21.10.4
port 40800
building initial quay.io/ceph-ci/ceph:37973791 image...
build
STEP 1: FROM quay.io/ceph-ci/ceph:master
STEP 2: COMMIT quay.io/ceph-ci/ceph:37973791
abe7892103f789d0030c4263d4e0a0d55f06e19f7dea61603e892968929aca90
INFO:cephadm:Verifying podman|docker is present...
INFO:cephadm:Verifying lvm2 is present...
INFO:cephadm:Verifying time synchronization is in place...
INFO:cephadm:Unit chronyd.service is enabled and running
INFO:cephadm:Repeating the final host check...
INFO:cephadm:podman|docker (/bin/podman) is present
INFO:cephadm:systemctl is present
INFO:cephadm:lvcreate is present
INFO:cephadm:Unit chronyd.service is enabled and running
INFO:cephadm:Host looks OK
INFO:root:Cluster fsid: 37973791-2508-40fe-85f5-f46afad11078
INFO:cephadm:Verifying IP 172.21.10.4 port 40800 ...
INFO:cephadm:Mon IP 172.21.10.4 is in CIDR network 172.21.0.0/20
INFO:cephadm:Extracting ceph user uid/gid from container image...
INFO:cephadm:Creating initial keys...
INFO:cephadm:Creating initial monmap...
INFO:cephadm:Creating mon...
INFO:cephadm:Waiting for mon to start...
INFO:cephadm:Waiting for mon...
INFO:cephadm:mon is available
INFO:cephadm:Assimilating anything we can from ceph.conf...
INFO:cephadm:Generating new minimal ceph.conf...
INFO:cephadm:Restarting the monitor...
INFO:cephadm:Setting mon public_network...
INFO:cephadm:Creating mgr...
INFO:cephadm:Wrote keyring to ./ceph.client.admin.keyring
INFO:cephadm:Wrote config to ./ceph.conf
INFO:cephadm:Waiting for mgr to start...
INFO:cephadm:Waiting for mgr...
INFO:cephadm:mgr not available, waiting (1/10)...
INFO:cephadm:mgr not available, waiting (2/10)...
INFO:cephadm:mgr is available
INFO:cephadm:Enabling cephadm module...
INFO:cephadm:Waiting for the mgr to restart...
INFO:cephadm:Waiting for mgr epoch 5...
INFO:cephadm:mgr epoch 5 is available
INFO:cephadm:Setting orchestrator backend to cephadm...
INFO:cephadm:Generating ssh key...
INFO:cephadm:Wrote public SSH key to to ./ceph.pub
INFO:cephadm:Adding key to root@localhost's authorized_keys...
INFO:cephadm:Adding host vossi04...
INFO:cephadm:Deploying mon service with default placement...
INFO:cephadm:Deploying mgr service with default placement...
INFO:cephadm:Deploying crash service with default placement...
INFO:cephadm:Enabling mgr prometheus module...
INFO:cephadm:Deploying prometheus service with default placement...
INFO:cephadm:Deploying grafana service with default placement...
INFO:cephadm:Deploying node-exporter service with default placement...
INFO:cephadm:Deploying alertmanager service with default placement...
INFO:cephadm:Enabling the dashboard module...
INFO:cephadm:Waiting for the mgr to restart...
INFO:cephadm:Waiting for mgr epoch 13...
INFO:cephadm:mgr epoch 13 is available
INFO:cephadm:Generating a dashboard self-signed certificate...
INFO:cephadm:Creating initial admin user...
INFO:cephadm:Fetching dashboard port number...
INFO:cephadm:Ceph Dashboard is now available at:

	     URL: https://vossi04.front.sepia.ceph.com:8443/
	    User: admin
	Password: dhwxi4aa4h

INFO:cephadm:You can access the Ceph CLI with:

	sudo ../src/cephadm/cephadm shell --fsid 37973791-2508-40fe-85f5-f46afad11078 -c ./ceph.conf -k ./ceph.client.admin.keyring

INFO:cephadm:Please consider enabling telemetry to help improve Ceph:

	ceph telemetry on

For more information see:

	https://docs.ceph.com/docs/master/mgr/telemetry/

INFO:cephadm:Bootstrap complete.

sudo ../src/script/cpatch -t quay.io/ceph-ci/ceph:37973791


```

File: src/script/cpatch [hunk 1]
```diff
@@ -47,6 +47,10 @@ while [ -n "$1" ]; do
 	    shift
 	    target="$1"
 	    ;;
+    --noall)
```
Comment (tchaikov) [inline src/script/cpatch:50]: please document what `--noall` is for in `usage()`.

File: src/script/cpatch [hunk 2]
```diff
@@ -47,6 +47,10 @@ while [ -n "$1" ]; do
...
+	    all=0
+        strip=0
```
Comment (tchaikov) [inline src/script/cpatch:52]: wrong indent.

File: src/script/cpatch (hunk 1 above)
Comment (tchaikov) [reply src/script/cpatch:50]: also, this change is not reflected in the commit message.

Comment (ideepika) [reply src/script/cpatch:50]: ack, thanks!

File: src/script/cpatch [hunk 3]
```diff
@@ -30,7 +30,7 @@ usage() {
     echo "  --push               push when done"
     echo "  --strip              strip binaries"
     echo "  --root-build         build image as root"
-    echo
+    echo "  --noall              skip all other operations and just check if build image is available"
```
Comment (tchaikov) [inline src/script/cpatch:33]: how does this script "check if build image is available"? could you paste a sample output here?

**Comment by ideepika** [comment]
sure!
```
[ideepika@vossi04 build]$ ../src/script/cpatch --target quay.io/ceph-ci/ceph
--noall
build
STEP 1: FROM quay.io/ceph-ci/ceph:master
STEP 2: COMMIT quay.io/ceph-ci/ceph
abe7892103f789d0030c4263d4e0a0d55f06e19f7dea61603e892968929aca90
```

**Comment by tchaikov** [comment]
jenkins test docs

**Pull request merged**